src/test/java/com/amazonaws/encryptionsdk/keyrings/KeyringTraceTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except
+ * in compliance with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.amazonaws.encryptionsdk.keyrings;
+
+import org.junit.Test;
+
+import static java.util.Collections.singleton;
+import static org.junit.Assert.assertEquals;
+
```

Comment:
Can we get some tests of the equality checks and tostrings?

Reply:
Added tests for equality. For toString, I'm a little hesitant to add tests, mainly because I'm using a library to generate the toString, so it's not really guaranteed (nor particularly important) how the formatting will end up if the library is updated.